File: MetaGeta/MetaGeta.DataStore/Templates/TVShowDataStoreTemplate.vb
Public Class TVShowDataStoreTemplate
    Implements IDataStoreTemplate

    Public Function GetDimensionNames() As String() Implements IDataStoreTemplate.GetDimensionNames
        Return New String() {SeriesTitle, EpisodeTitle, EpisodeNumber, SeasonNumber, PartNumber, _
                             SeriesID, EpisodeID, SeriesDescription, EpisodeDescription, EpisodeBanner, _
                             CRC32, Group, AudioCodec, VideoCodec, Resolution, PlayTime}
    End Function

    Public Function GetColumnNames() As String() Implements IDataStoreTemplate.GetColumnNames
        Return New String() {SeriesTitle, EpisodeTitle, EpisodeNumber, SeasonNumber, PartNumber, _
                             CRC32, Group, AudioCodec, VideoCodec, Resolution, PlayTime}
    End Function

    Public Function GetPluginTypeNames() As String() Implements IDataStoreTemplate.GetPluginTypeNames
        Return New String() { _
                             "MetaGeta.DataStore.FileInfoTaggingPlugin, MetaGeta.DataStore", _
                             "MetaGeta.TVShowPlugin.EducatedGuessImporter, TVShowPlugin", _
                             "MetaGeta.DirectoryFileSourcePlugin.DirectoryFileSourcePlugin, DirectoryFileSourcePlugin", _
                             "MetaGeta.MediaInfoPlugin.MediaInfoPlugin, MediaInfoPlugin", _
                             "MetaGeta.TVDBPlugin.TVDBPlugin, TVDBPlugin", _
                             "TranscodePlugin.TranscodePlugin, TranscodePlugin", _
                             "TranscodePlugin.Mp4TagWriterPlugin, TranscodePlugin" _
                            }
    End Function

    Public Function GetName() As String Implements IDataStoreTemplate.GetName
        Return "TVShow"
    End Function

#Region "Constants"
    Public Const SeriesTitle As String = "SeriesTitle"
    Public Const EpisodeTitle As String = "EpisodeTitle"

    Public Const EpisodeNumber As String = "EpisodeNumber"
    Public Const SeasonNumber As String = "SeasonNumber"
    Public Const PartNumber As String = "PartNumber"

    Public Const SeriesID As String = "SeriesID"
    Public Const EpisodeID As String = "EpisodeID"
    Public Const EpisodeFirstAired As String = "EpisodeFirstAired"

    Public Const EpisodeBanner As String = "EpisodeBanner"

    Public Const SeriesDescription As String = "SeriesDescription"
    Public Const EpisodeDescription As String = "EpisodeDescription"

    Public Const CRC32 As String = "CRC32"
    Public Const Group As String = "Group"

    Public Const Format As String = "Format"

    Public Const AudioCodec As String = "AudioCodec"
    Public Const VideoCodec As String = "VideoCodec"
    Public Const VideoCodecProfile As String = "VideoCodecProfile"

    Public Const Resolution As String = "Resolution"
    Public Const PlayTime As String = "PlayTime"
    Public Const FrameCount As String = "FrameCount"
    Public Const FrameRate As String = "FrameRate"
    Public Const VideoWidthPx As String = "VideoWidthPx"
    Public Const VideoHeightPx As String = "VideoHeightPx"
    Public Const VideoDisplayAspectRatio As String = "VideoDisplayAspectRatio"

    Public Const iPod5GCompatible As String = "iPod5GCompatible"
    Public Const iPodClassicCompatible As String = "iPodClassicCompatible"
    Public Const iPhoneCompatible As String = "iPhoneCompatible"
#End Region
End Class

File: MetaGeta/MetaGeta.DataStore/Templates/TVShowDataStoreTemplate.cs
// Copyright 2009 Will Thomas
// 
// This file is part of MetaGeta.
// 
// MetaGeta is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// MetaGeta is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with MetaGeta. If not, see <http://www.gnu.org/licenses/>.
namespace MetaGeta.DataStore {
    public class TVShowDataStoreTemplate : IDataStoreTemplate {
        public const string iPhoneCompatible = "iPhoneCompatible";

        #region IDataStoreTemplate Members

        public string[] GetDimensionNames() {
            return new[] {
                             SeriesTitle,
                             EpisodeTitle,
                             EpisodeNumber,
                             SeasonNumber,
                             PartNumber,
                             SeriesID,
                             EpisodeID,
                             SeriesDescription,
                             EpisodeDescription,
                             EpisodeBanner,
                             CRC32,
                             Group,
                             AudioCodec,
                             VideoCodec,
                             Resolution,
                             PlayTime
                         };
        }

        public string[] GetColumnNames() {
            return new[] {
                             SeriesTitle,
                             EpisodeTitle,
                             EpisodeNumber,
                             SeasonNumber,
                             PartNumber,
                             CRC32,
                             Group,
                             AudioCodec,
                             VideoCodec,
                             Resolution,
                             PlayTime
                         };
        }

        public string[] GetPluginTypeNames() {
            return new[] {
                             "MetaGeta.DataStore.FileInfoTaggingPlugin, MetaGeta.DataStore",
                             "MetaGeta.TVShowPlugin.EducatedGuessImporter, TVShowPlugin",
                             "MetaGeta.DirectoryFileSourcePlugin.DirectoryFileSourcePlugin, DirectoryFileSourcePlugin",
                             "MetaGeta.MediaInfoPlugin.MediaInfoPlugin, MediaInfoPlugin",
                             "MetaGeta.TVDBPlugin.TVDBPlugin, TVDBPlugin",
                             "TranscodePlugin.TranscodePlugin, TranscodePlugin",
                             "TranscodePlugin.Mp4TagWriterPlugin, TranscodePlugin"
                         };
        }

        public string GetName() {
            return "TVShow";
        }

        #endregion

        #region "Constants"

        public const string AudioCodec = "AudioCodec";
        public const string CRC32 = "CRC32";
        public const string EpisodeBanner = "EpisodeBanner";

        public const string EpisodeDescription = "EpisodeDescription";
        public const string EpisodeFirstAired = "EpisodeFirstAired";
        public const string EpisodeID = "EpisodeID";
        public const string EpisodeNumber = "EpisodeNumber";
        public const string EpisodeTitle = "EpisodeTitle";

        public const string Format = "Format";
        public const string FrameCount = "FrameCount";
        public const string FrameRate = "FrameRate";
        public const string Group = "Group";
        public const string iPod5GCompatible = "iPod5GCompatible";
        public const string iPodClassicCompatible = "iPodClassicCompatible";
        public const string PartNumber = "PartNumber";
        public const string PlayTime = "PlayTime";
        public const string Resolution = "Resolution";
        public const string SeasonNumber = "SeasonNumber";
        public const string SeriesBanner = "SeriesBanner";
        public const string SeriesDescription = "SeriesDescription";
        public const string SeriesID = "SeriesID";
        public const string SeriesPoster = "SeriesPoster";
        public const string SeriesTitle = "SeriesTitle";
        public const string VideoCodec = "VideoCodec";

        public const string VideoCodecProfile = "VideoCodecProfile";
        public const string VideoDisplayAspectRatio = "VideoDisplayAspectRatio";
        public const string VideoHeightPx = "VideoHeightPx";
        public const string VideoWidthPx = "VideoWidthPx";

        #endregion
    }
}